web/src/main/java/no/nav/modiapersonoversikt/rest/persondata/PersondataResult.kt

```diff
@@ -51,6 +51,14 @@ sealed class PersondataResult<T>(val system: InformasjonElement) {
         }
     }
 
+    fun <S> flatMap(block: (t: PersondataResult<T>) -> PersondataResult<S>): PersondataResult<S> {
+        return when (this) {
+            is Failure<*> -> this as PersondataResult<S>
+            is NotRelevant<*> -> this as PersondataResult<S>
+            is Success<T> -> block(this.value as PersondataResult<T>)
```

Comment:
Denne castingen vil feile siden `this.value` vil ha typen `T`

Vil sannsynligvis ha noe alà;
```kotlin
is Success<T> -> block(this)
```